File: aspnet-in-practice/16/VB/ParallelFX/Chapter15.ParallelFX/Chapter15.ParallelFX/Code/Providers/SecondProvider.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Threading

Public Class SecondProvider
	Implements IFlightPriceProvider
	Public Function GetFlightPrice(ByVal flightNumber As String) As FlightPriceResult Implements IFlightPriceProvider.GetFlightPrice
		' let's simulate a call to a remote resource
		Thread.Sleep(3000)

		' and return a fixed value
		Dim result As New FlightPriceResult()
		result.FlightNumber = flightNumber
		result.Price = 260
		Return result
	End Function

End Class

File: aspnet-in-practice/16/CS/ThreadPool/App_Code/Providers/SecondProvider.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Threading;

public class SecondProvider: IFlightPriceProvider
{
	public FlightPriceResult GetFlightPrice(string flightNumber)
	{
		// let's simulate a call to a remote resource
		Thread.Sleep(3000);

		// and return a fixed value
		return new FlightPriceResult() { FlightNumber = flightNumber, Price = 260 };
	}

}